policy-controller/k8s/status/src/routes/grpc.rs

```diff
@@ -0,0 +1,173 @@
+use super::BackendReference;
+use linkerd_policy_controller_k8s_api::gateway as k8s_gateway_api;
+
+pub(crate) fn make_backends(
+    namespace: &str,
+    backends: impl Iterator<Item = k8s_gateway_api::GrpcRouteBackendRef>,
+) -> Vec<BackendReference> {
+    backends
+        .map(|backend_ref| BackendReference::from_backend_ref(&backend_ref.inner, namespace))
+        .collect()
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::index::POLICY_API_GROUP;
+    use linkerd_policy_controller_k8s_api::{self as k8s_core_api, gateway as k8s_gateway_api};
+
+    fn default_grpc_backends(
```

Comment:
another quick and nitpicky thought, does it make sense for this to return an Option if we never return `None`? An alternative would be to return a vec here and wrap it in an option when creating the backendrefs from route

```
backend_refs: Some(default_grpc_backends(vec![
                            k8s_gateway_api::BackendObjectReference {
                                group: None,
                                kind: None,
                                name: "ref-1".to_string(),
                                namespace: Some("default".to_string()),
                                port: None,
                            }),
```

Reply:
Certainly we could, but the purpose of this function is to cut down on boilerplate and make it more succinct to construct route rules so I think it makes sense to factor the `Some` into this function even though it doesn't carry any semantic value.